### PR TITLE
changes to dataloader and evaluate_model

### PR DESCRIPTION
**Pull request by benntay1**

### CODE REVIEW

File: src/data_loader.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 import random
 import pandas as pd
 import h5py
-
+import ast
```
Comment (teojunjie) [inline src/data_loader.py:20]: what is ast?

File: src/data_loader.py [hunk 2]
```diff
@@ -76,18 +82,24 @@ def __iter__(self):
...
                 # target shape
                 target_shape = (720, 1280)
                 divide = 2**num_pool
                 gt_target_shape = (720//divide, 1280//divide)
 
                 # resizing with cv2
-                img_resized = cv2.resize(img, target_shape, interpolation = cv2.INTER_CUBIC)
-                gt_resized = cv2.resize(den, gt_target_shape, interpolation = cv2.INTER_CUBIC)
+                img_resized = cv2.resize(img, target_shape, interpolation = cv2.INTER_LINEAR)
+                gt_resized = cv2.resize(den, gt_target_shape, interpolation = cv2.INTER_LINEAR) * (4**num_pool)
```
Comment (teojunjie) [inline src/data_loader.py:94]: Ok so we no longer multiplying by 16 instead?

File: src/evaluate_model.py [hunk 3]
```diff
@@ -6,22 +6,76 @@
...
-    mse = np.sqrt(mse/data_loader.get_num_samples())
-    return mae,mse
\ No newline at end of file
+        # updating the values
+        MAEcrowddensity[crowd_density] += abs(gt_count-et_count)
+        MSEcrowddensity[crowd_density] += ((gt_count-et_count)*(gt_count-et_count))
+        MAEweather[weather] += abs(gt_count-et_count)
+        MSEweather[weather] += ((gt_count-et_count)*(gt_count-et_count))
+        MAE += abs(gt_count-et_count)
+        MSE += ((gt_count-et_count)*(gt_count-et_count))
```
Comment (teojunjie) [inline src/evaluate_model.py:46]: Is this the correct equation for mse?

**Review by teojunjie — COMMENTED** [review]
Couple of comments

File: src/data_loader.py (hunk 1 above)
Comment (benntay1) [reply src/data_loader.py:20]: needed to import this library, cos i cant store a dictionary into the hdf file for some reason
so I converted the dictionary to string, then `ast` is to convert the string back into a dictionary within the code

File: src/data_loader.py (hunk 2 above)
Comment (benntay1) [reply src/data_loader.py:94]: this one im actually not too sure eh, I'm still reading up on how to handle these different image dimensions cos the density maps get distorted when resizing

i think for now can leave it first, once I get some solid insights I will update the resizing operations again

File: src/data_loader.py (hunk 1 above)
Comment (teojunjie) [reply src/data_loader.py:20]: Did u add this to the docker file? Or do we have to add it ourself

File: src/data_loader.py (hunk 2 above)
Comment (teojunjie) [reply src/data_loader.py:94]: Noted!

File: src/evaluate_model.py (hunk 3 above)
Comment (teojunjie) [reply src/evaluate_model.py:46]: Alright!

**Review by teojunjie — APPROVED** [review]

**Pull request merged**